src/app/components/canvas/canvas.component.ts

```diff
@@ -88,6 +89,9 @@ export class CanvasComponent {
           board.permissions.allowStudentMoveAny ? this.lockPostsMovement(false) : this.lockPostsMovement(true)
           board.bgImage ? this.updateBackground(board.bgImage.url, board.bgImage.imgSettings) : null
           this.updateShowAddPost(this.board.permissions)
+          if(!((this.user.role =="student" && this.board.permissions.showAuthorName) || this.user.role =="teacher")){
```

Comment:
Can you extract this logic into smaller chunks, similar to my suggestions in the locking PR?